kotlin-inject-compiler/core/src/main/kotlin/me/tatarka/inject/compiler/TypeCollector.kt

```diff
@@ -253,7 +252,7 @@ class TypeCollector(private val provider: AstProvider, private val options: Opti
         return typeInfoCache.getOrPut(astClass.toString()) {
             val isComponent = astClass.isComponent()
 
-            val providesMethods = mutableListOf<AstMember>()
+            val providesMethods = mutableListOf<Pair<AstMember, AstType?>>()
```

Comment:
@evant would you prefer a custom type instead of `Pair`?

Reply:
no strong opinions